clients/unshaded/src/main/java/tachyon/client/block/BufferedBlockInStream.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.block;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.Constants;
+import tachyon.client.ClientContext;
+import tachyon.conf.TachyonConf;
+import tachyon.util.io.BufferUtils;
+
+/**
+ * Provides a stream API to read a block from Tachyon. An instance extending this class can be
```

Comment:
The javadoc here is copied from original `BlockInStream`, does it still make sense for this new abstract class?